test/replaytests/replay_test.go

```diff
@@ -431,6 +431,20 @@ func (s *replayTestSuite) TestMultipleUpdates() {
 	s.NoError(err)
 }
 
+func (s *replayTestSuite) TestResetWithUpdateAccepted() {
+	replayer := worker.NewWorkflowReplayer()
+	replayer.RegisterWorkflow(CounterWorkflow)
+	err := replayer.ReplayWorkflowHistoryFromJSONFile(ilog.NewDefaultLogger(), "update-reset-accepted.json")
+	s.NoError(err)
+}
+
+func (s *replayTestSuite) TestResetWithUpdateRejected() {
+	replayer := worker.NewWorkflowReplayer()
+	replayer.RegisterWorkflow(CounterWorkflow)
+	err := replayer.ReplayWorkflowHistoryFromJSONFile(ilog.NewDefaultLogger(), "update-reset-rejected.json")
+	s.NoError(err)
```

Comment:
So these tests just check that replay does not error, but do not make any assertions about the state attained by replay, is that right? Should we make some assertions (e.g. assertions about log output? Is that the point of the logging in the workflow and validator?)? Otherwise what is the point of the `CounterWorkflow` body implementation? (Sorry, I'm new to these tests).

Reply:
> but do not make any assertions about the state attained by replay, is that right?

They assert the replay succeeded which means the workflow took the same steps it took on initial execution which is dependent on the state of the workflow. 

We could remove the logs, they don't serve any purpose on replay.